File: 98/Go/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isValidBST(root *TreeNode) bool {
	arr := make([]int, 0)
	var spread func(root *TreeNode)
	spread = func(root *TreeNode) {
		if root == nil {
			return
		}
		spread(root.Left)
		arr = append(arr, root.Val)
		spread(root.Right)
	}
	spread(root)
	flag := true
	for i := 0; i < len(arr)-1 && flag; i++ {
		//如果有等于号，则依然return false
		if arr[i] >= arr[i+1] {
			flag = false
		}
	}
	return flag
}
func main() {

}

File: 98/Cpp/main.cpp
#include<iostream>
#include<vector>
#include<climits>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode* left, TreeNode* right) : val(x), left(left), right(right) {}
};
/**
 * @brief 
 * funciton1
 */
// class Solution {
// public:
//     vector<int> nums;
//     bool isValidBST(TreeNode* root) {
//         getVector(nums, root);
//         for (int i = 1 ; i < nums.size(); i++ ) {
//             if (nums[i] <= nums[i-1]) {
//                 return false;
//             }
//         }
//         return true;
//     }
//     void getVector(vector<int>& nums, TreeNode* node) {
//         if ( node == nullptr ) {
//             return;
//         }
//         getVector(nums, node->left);
//         nums.push_back(node->val);
//         getVector(nums, node->right);
//     }
// };


class Solution {
public:
    bool helper(TreeNode* node, long long lower, long long upper) {
        if ( node == nullptr ) {
            return true;
        }

        if ( (node -> val) <= lower || (node -> val) >= upper) {
            return false;
        }

        return helper(node->left, lower, node -> val) &&
            helper(node -> right, node -> val, upper);
    }
    bool isValidBST(TreeNode* root) {
        return helper(root, LLONG_MIN, LLONG_MAX);
    }
};